relayer/chains/substrate/query.go

```diff
@@ -479,6 +491,57 @@ func (sp *SubstrateProvider) QueryHeaderAtHeight(ctx context.Context, height int
 	return sp.constructBeefyHeader(blockHash, nil)
 }
 
+// QueryIBCHeader returns the result of QueryLatestIBCHeader. It is only applicable when creating clients.
+// QueryIBCHeaderOverBlocks should be used in the chain processor when querying headers to update the client.
+func (sp *SubstrateProvider) QueryIBCHeader(ctx context.Context, h int64) (provider.IBCHeader, error) {
+	if h <= 0 {
+		return nil, fmt.Errorf("must pass in valid height, %d not valid", h)
+	}
+
+	latestFinalizedHeight, err := sp.QueryLatestHeight(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if h > latestFinalizedHeight {
+		return nil, fmt.Errorf("queried height is not finalized")
+	}
+
+	return sp.QueryLatestIBCHeader()
+}
+
+// QueryLatestIBCHeader returns the IBCHeader at the latest finalized relay chain height. It uses the relay chain height
+// which is set in the SubstrateProvider when the QueryLatestHeight method is called.
+func (sp *SubstrateProvider) QueryLatestIBCHeader() (provider.IBCHeader, error) {
+	var relayChainHeight uint64
```

Comment:
I don't see a reason why this variable needs to be declared before initialization down on L521